packages/apple-pay-integration/src/apple-pay-button-strategy.ts

```diff
@@ -103,24 +115,60 @@ export default class ApplePayButtonStrategy implements CheckoutButtonStrategy {
         return button;
     }
 
-    private _handleWalletButtonClick(event: Event) {
+    private async _handleWalletButtonClick(event: Event) {
         event.preventDefault();
 
-        const state = this._paymentIntegrationService.getState();
-        const cart = state.getCartOrThrow();
-        const config = state.getStoreConfigOrThrow();
-        const checkout = state.getCheckoutOrThrow();
-
         if (!this._paymentMethod || !this._paymentMethod.initializationData) {
             throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
         }
 
-        const request = this._getBaseRequest(cart, checkout, config, this._paymentMethod);
-        const applePaySession = this._sessionFactory.create(request);
+        if (
+            this._buyNowInitializeOptions &&
+            typeof this._buyNowInitializeOptions.getBuyNowCartRequestBody === 'function'
+        ) {
+            const {
+                countryCode,
+                currencyCode,
+                merchantCapabilities,
+                supportedNetworks,
+                storeName,
+            } = this._paymentMethod.initializationData;
+
+            const request: ApplePayJS.ApplePayPaymentRequest = {
+                countryCode,
+                currencyCode,
+                supportedNetworks,
+                merchantCapabilities,
+                total: { label: '', amount: '0', type: 'pending' },
+                requiredBillingContactFields: ['postalAddress'],
+                requiredShippingContactFields: ['email', 'phone'],
+            };
```

Comment:
Can we create a method, `this._getBuyNowCartRequest`, to do this?
Also, why `total: { label: '', amount: '0', type: 'pending' },` is different from the base request?

Reply:
added this._getBuyNowCartRequest, and added label: storeName to total object